nomad/job_endpoint.go

```diff
@@ -905,6 +905,9 @@ func (j *Job) Scale(args *structs.JobScaleRequest, reply *structs.JobRegisterRes
 	if args.Error && args.Count != nil {
 		return structs.NewErrRPCCoded(400, "scaling action should not contain count if error is true")
 	}
+	if *args.Count < 0 {
```

Comment:
Should also add validation for `args.Count` nilness when `args.Error` is false?  Also, this needs to gate on its nilness?

Reply:
Added the nilness gate on this validation.

On the first point; I believe it is possible a scaling event could be submitted which has a nil count (so merely informational) whose message isn't an error.